docs/design/settings/extensions/settings-extension-example.md

```diff
@@ -0,0 +1,105 @@
+# Settings Extensions Example
+This document walks through a typical use-case for Bottlerocket's settings extension system.
+Technical details of the design of the settings extension system are described elsewhere.
+
+## Background
+Suppose you are creating a Bottlerocket variant which adds some new daemon software that you've created.
+The daemon is configured via a YAML configuration file with a web address.
+When the daemon starts, it parses its configuration file for the target address and then begins periodically querying that address for updates.
+
+You'd like the following behavior to be accomplished:
+* You can configure the daemon using the Bottlerocket settings API, the value will be stored in `settings.mydaemon.query-url`.
+* Whenever the daemon is reconfigured using the Bottlerocket API, it is automatically restarted with the new configuration.
+
+In order to do this, we need to create the following resources in our Bottlerocket variant:
+* A setting extension called `mydaemon` to add the new value `settings.mydaemon.query-url` to the API.
```

Comment:
Should it be called mydaemon-settings to distinguish it from the mydaemon daemon binary?

```suggestion
* A setting extension called `mydaemon-settings` to add the new value `settings.mydaemon.query-url` to the API.
```